File: java/70-Climbing-Stairs.java
class Solution {
    public int climbStairs(int n) {
        
        if(n < 2){
            return 1;
        }
        
        int [] dp = new int [n];
        dp[n - 1] = 1;
        dp[n - 2] = 2;
        
        for(int i = n - 3; i >= 0; i--){
            dp[i] = dp[i + 1] + dp[i + 2];
        }
        
        return dp[0];
    }
}

File: swift/70-Climbing-Stairs.swift
class Solution {
    func climbStairs(_ n: Int) -> Int {
        if n <= 3 {
            return n
        }
        var n1 = 2
        var n2 = 3

        for i in 4...n {
            let temp = n1 + n2
            n1 = n2
            n2 = temp
        }
        return n2
    }
}